edgedb/options.py

```diff
@@ -0,0 +1,291 @@
+import abc
+import enum
+import random
+from collections import namedtuple
+
+from . import errors
+
+
+_RetryRule = namedtuple("_RetryRule", ["attempts", "backoff"])
+
+
+def default_backoff(attempt):
+    return (2 ** attempt) * 0.1 + random.randrange(100) * 0.001
+
+
+class RetryCondition:
+    """Specific condition to retry on for fine-grained control"""
+    SerializationError = enum.auto()
+    Deadlock = enum.auto()
```

Comment:
Merge this into `TransactionConflictError` as discussed in edgedb/edgedb-go#123

Reply:
`Conflict` is a bit too generic.  Can we say `TransactionConflict`?

Reply:
Well, yes, since we have `UNLESS CONFLICT` one may imply that this is related.